lib/libzpool/uzfs_mgmt.c

```diff
@@ -462,6 +463,16 @@ uzfs_create_dataset(spa_t *spa, char *ds_name, uint64_t vol_size,
 	return (error);
 }
 
+int
+is_rebuild_cloned_volume(const char *ds_name)
+{
+	size_t ds_len  = strlen(ds_name);
+	size_t pattern_len = strlen(REBUILD_SNAPSHOT_CLONENAME);
+
+	return (strcmp(ds_name + (ds_len - pattern_len),
```

Comment:
we have to check for lengths

Reply:
and, we have to check that "REBUILD_SNAPSHOT_CLONENAME" is at the tail of the string

Reply:
We are checking that pattern should be at the end of ds_name. We are advancing ds_name to the start of index where pattern should start in ds_name. Am I missing something here sir ?

Reply:
hmm.. we need to consider the case of ds_len being less than pattern_len, so that, (ds_len - pattern_len) won't go negative.
And, logic of advancing by pattern length is right, sir, to check for the tail..